ReleaseProcessAutomation.Tests/Steps/Releases/ReleaseRcStepTests.cs

```diff
@@ -0,0 +1,231 @@
+          _msBuildInvokerMock.Object,
+          _continueAlphaBetaMock.Object,
+          _consoleStub.Object);
+
+      Assert.That(
+          () => rcStep.Execute(new SemanticVersion(), "", true, false, "hotfix/v1.3.5"),
+          Throws.Nothing);
+
+      _continueAlphaBetaMock.Verify(_ => _.Execute(It.IsAny<SemanticVersion>(), It.IsAny<string>(), It.IsAny<string>(), false), Times.Never);
+    }
+
+
+    [Test]
+    public void Execute_WithoutErrors_CallsNextStep ()
+    {
+      _gitClientStub.Setup(_ => _.IsWorkingDirectoryClean()).Returns(true);
+      _gitClientStub.Setup(_ => _.IsCommitHash("")).Returns(true);
+      _gitClientStub.Setup(_ => _.IsOnBranch("release/")).Returns(true);
+      _gitClientStub.Setup(_ => _.GetCurrentBranchName()).Returns("branchName");
+      _gitClientStub.Setup(_ => _.DoesBranchExist("prerelease/v0.0.0")).Returns(false);
+      var nextPossibleVersions = new SemanticVersion().GetNextPossibleVersionsHotfix();
+      _continueAlphaBetaMock.Setup(_ => _.Execute(It.IsAny<SemanticVersion>(), It.IsAny<string>(), It.IsAny<string>(), false)).Verifiable();
+
+      _inputReaderMock.Setup(_ => _.ReadVersionChoice(It.IsAny<string>(), nextPossibleVersions)).Returns(new SemanticVersion());
+
+      var rcStep = new ReleaseRCStep(
+          _gitClientStub.Object,
+          _config,
+          _inputReaderMock.Object,
+          _ancestorStub.Object,
+          _msBuildInvokerMock.Object,
+          _continueAlphaBetaMock.Object,
+          _consoleStub.Object);
+
+      Assert.That(
+          () => rcStep.Execute(new SemanticVersion(), "", false, false, "hotfix/v1.3.5"),
+          Throws.Nothing);
+      _continueAlphaBetaMock.Verify();
+    }
+
+    [Test]
+    public void Execute_WithoutErrors_CallsMBuildInvokeAndCommit()
+    {
+      _gitClientStub.Setup(_ => _.IsWorkingDirectoryClean()).Returns(true);
+      _gitClientStub.Setup(_ => _.IsCommitHash("")).Returns(true);
+      _gitClientStub.Setup(_ => _.IsOnBranch("release/")).Returns(true);
+      _gitClientStub.Setup(_ => _.GetCurrentBranchName()).Returns("branchName");
+      _gitClientStub.Setup(_ => _.DoesBranchExist("prerelease/v0.0.0")).Returns(false);
+      var nextPossibleVersions = new SemanticVersion().GetNextPossibleVersionsHotfix();
+      _msBuildInvokerMock.Setup(_ => _.CallMSBuildStepsAndCommit(MSBuildMode.PrepareNextVersion, new SemanticVersion())).Verifiable();
+      _inputReaderMock.Setup(_ => _.ReadVersionChoice(It.IsAny<string>(), nextPossibleVersions)).Returns(new SemanticVersion());
+
+      var rcStep = new ReleaseRCStep(
+          _gitClientStub.Object,
+          _config,
+          _inputReaderMock.Object,
+          _ancestorStub.Object,
+          _msBuildInvokerMock.Object,
+          _continueAlphaBetaMock.Object,
+          _consoleStub.Object);
+
+      Assert.That(
+          () => rcStep.Execute(new SemanticVersion(), "", false, false, "hotfix/v1.3.5"),
+          Throws.Nothing);
+      _msBuildInvokerMock.Verify();
```

Comment:
It's not wrong to do it this way, but you could just call the Execute method and then mock.Verify. No need to assert that no exception is thrown since this would fail the test anyway and the mock.Verify makes it clear what we are asserting.